frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseRepositoryAction.java

```diff
@@ -0,0 +1,292 @@
+import com.virtuslab.gitmachete.frontend.actions.navigation.CheckoutPreviousAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.OverrideForkPointOfCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.PullCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.PushCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.RefreshStatusAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.ResetCurrentToRemoteAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.SlideOutCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.SyncCurrentToParentByRebaseAction;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.gitmachete.backend.api.ConfirmedNonRoot;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public class TraverseRepositoryAction extends BaseGitMacheteRepositoryReadyAction implements IExpectsKeySelectedBranchName {
+
+  @Untainted
+  @I18nFormat({I18nConversionCategory.GENERAL})
+  final String actionIdFormatString = getNonHtmlString("action.GitMachete.TraverseRepositoryAction.invoked-action.id");
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  public static final String SHOW_PULL_APPROVAL = "git-machete.pull.approval.show";
+  public static final String SHOW_PUSH_APPROVAL = "git-machete.push.approval.show";
+  public static final String SHOW_TRAVERSE_APPROVAL = "git-machete.traverse.approval.show";
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+    presentation.setDescription(getNonHtmlString("action.GitMachete.TraverseRepositoryAction.description.text"));
+
+    val repositorySnapshot = getGitMacheteRepositorySnapshot(anActionEvent);
+    val branchLayout = repositorySnapshot != null ? repositorySnapshot.getBranchLayout() : null;
+
+    if (branchLayout == null || branchLayout.getRootEntries().isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.TraverseRepositoryAction.description.empty-layout"));
+      return;
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val repository = getSelectedGitRepository(anActionEvent);
+    val branchLayout = getBranchLayout(anActionEvent);
+
+    if (branchLayout != null && branchLayout.getRootEntries().nonEmpty() && repository != null) {
+      boolean traverseApproved = true;
+      if (PropertiesComponent.getInstance().getBoolean(SHOW_TRAVERSE_APPROVAL, /* defaultValue */ true)) {
+
+        val traverseApprovalDialogBuilder = MessageDialogBuilder.okCancel(
```

Comment:
TBH instead of asking whether to _start traversal_ (for which in 99%+ cases the answer gonna be _yes_), I'd more specifically ask whether to check out the first branch to be worked upon